Add ambient helm installation guide (#13871)
istio/istio.io: content/en/docs/ops/ambient/usage/helm-installation/index.md

---

> title: Installing Ambient Mesh with Helm

Reviewer: ```suggestion
title: Install Ambient Mesh with Helm
```

Reviewer: Can we have a separate Install section under Operations>Ambient Mesh, and then move this under there. Also it would make sense to remove the other empty Installation Guide page: https://deploy-preview-13871--preliminary-istio.netlify.app/latest/docs/ops/ambient/usage/install/

Author: Makes sense, I'll fix it. Thanks!

Author: @kfaseela Could you check this new section format? https://deploy-preview-13871--preliminary-istio.netlify.app/latest/docs/ops/ambient/install/

{{< text syntax=bash snip_id=install_discovery >}}
$ helm install istiod istio/istiod --namespace istio-system \
  -f @manifests/charts/istio-control/istio-discovery/ambient-values.yaml@
{{< /text >}}

### Installing the ztunnel component

The `ztunnel` chart installs the ztunnel DaemonSet, which is the node-proxy component of ambient.

{{< text syntax=bash snip_id=install_ztunnel >}}
$ helm install ztunnel istio/ztunnel -n istio-system
{{< /text >}}

## Configuration

To view support configuration options and documentation, run:

{{< text syntax=bash >}}
$ helm show values istio/istiod
{{< /text >}}

## Verifying the Installation

### Verifying the workload status

After installing all the components, you can check the Helm deployment status with:

{{< text syntax=bash snip_id=show_components >}}
$ helm list -n istio-system
{{< /text >}}

You can check the status of the deployed pods with:

{{< text syntax=bash snip_id=check_pods >}}
$ kubectl get pods -n istio-system
{{< /text >}}

### Verifying with the Sample Application

After installing ambient with Helm, you can follow
[Deploy the sample application](/docs/ops/ambient/getting-started/#bookinfo)

> guide to deploy the sample application and ingress gateways, and follow the
> [Adding your application to ambient](/docs/ops/ambient/getting-started/#addtoambient)
> to add your application to ambient.

Reviewer: ```suggestion
guide to deploy the sample application and ingress gateways, and then you can
[add your application to ambient](/docs/ops/ambient/getting-started/#addtoambient).
```

    {{< /text >}}

1. (Optional) Delete any Istio gateway chart installations:

    {{< text syntax=bash >}}
    $ helm delete istio-ingress -n istio-ingress
    $ kubectl delete namespace istio-ingress
    {{< /text >}}

1. Delete Istio CNI chart:

    {{< text syntax=bash snip_id=delete_cni >}}
    $ helm delete istio-cni -n istio-system
    {{< /text >}}

1. Delete Istio ztunnel chart:

    {{< text syntax=bash snip_id=delete_ztunnel >}}
    $ helm delete ztunnel -n istio-system
    {{< /text >}}

1. Delete Istio discovery chart:

    {{< text syntax=bash snip_id=delete_discovery >}}
    $ helm delete istiod -n istio-system
    {{< /text >}}

1. Delete Istio base chart:

    {{< tip >}}
    By design, deleting a chart via Helm doesn't delete the installed Custom
    Resource Definitions (CRDs) installed via the chart.
    {{< /tip >}}

    {{< text syntax=bash snip_id=delete_base >}}
    $ helm delete istio-base -n istio-system
    {{< /text >}}

1. Delete the `istio-system` namespace:

    {{< text syntax=bash snip_id=delete_system_namespace >}}
    $ kubectl delete namespace istio-system
    {{< /text >}}

> ## Uninstall stable revision label resources

Reviewer: Do we have to specify revision related commands here? I cannot see anything related to this in the install section. and isn't this same as the usual helm revision commands, and becomes a repitition

Author: I think we can just remove this section.
